### PR TITLE
updated checkout UI and cleaned up validations

### PR DESCRIPTION
**Pull request by AmyBeall**

### CODE REVIEW

File: app/views/spree/checkout/payment/_affirm.html.erb [hunk 1]
```diff
@@ -173,6 +167,15 @@
           }
         });
       });
+
+      /*****************************************************\
+          remove the payment source empty box from UI
+      \*****************************************************/
+      $(document).on('click', '#checkout_form_payment'
+        , function (event) {
+          if($('div[data-hook="checkout_payment_step"] input[type="radio"]:checked')[0].parentNode.outerText === "Affirm Monthly Payments") $('.payment-sources').hide();
```
Comment (kkong87) [inline app/views/spree/checkout/payment/_affirm.html.erb:176]: This looks like it hides everything with a class name of "payment-sources" is that correct?

File: app/controllers/spree/affirm_controller.rb [hunk 2]
```diff
@@ -24,31 +24,7 @@ def confirm
...
-            order.ship_address.phone = phone
-
-          when :billing_email
-            order.email = _affirm_checkout.details["billing"]["email"]
-
-          end
-        end
-
-        order.save
-      end
```
Comment (kkong87) [inline app/controllers/spree/affirm_controller.rb:51]: Is there a reason we don't need all this anymore?

File: app/models/spree/affirm_checkout.rb [hunk 3]
```diff
@@ -3,8 +3,6 @@ class AffirmCheckout < ActiveRecord::Base
     belongs_to :payment_method
     belongs_to :order
 
-    validate :validate_checkout_matches_order
```
Comment (kkong87) [inline app/models/spree/affirm_checkout.rb:6]: Is there another validation?

Comment (AmyBeall) [reply app/models/spree/affirm_checkout.rb:6]: We have the standard price validation in another price. This validation was technically removed a while back, I'm just cleaning up unused functions.

File: app/controllers/spree/affirm_controller.rb (hunk 2 above)
Comment (AmyBeall) [reply app/controllers/spree/affirm_controller.rb:51]: Same as below, it wasn't being used. This was checking the address before sending it through so it didn't quite make sense

File: app/views/spree/checkout/payment/_affirm.html.erb (hunk 1 above)
Comment (AmyBeall) [reply app/views/spree/checkout/payment/_affirm.html.erb:176]: That is correct. Payment sources is a div that is not needed for the Affirm payment method

**Comment by kkong87** [comment]
LGTM

**Pull request merged**